src/main/java/no/nav/familie/integrasjoner/personopplysning/PersonopplysningerService.kt

```diff
@@ -0,0 +1,81 @@
+import no.nav.tjeneste.virksomhet.person.v3.informasjon.NorskIdent
+import no.nav.tjeneste.virksomhet.person.v3.informasjon.Periode
+import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentPersonRequest
+import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentPersonResponse
+import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentPersonhistorikkRequest
+import org.slf4j.LoggerFactory
+import org.springframework.cache.annotation.Cacheable
+import org.springframework.stereotype.Service
+import org.springframework.web.context.annotation.ApplicationScope
+import java.time.LocalDate
+import no.nav.tjeneste.virksomhet.person.v3.informasjon.PersonIdent as TpsPersonIdent
+
+@Service
+@ApplicationScope
+class PersonopplysningerService(private val personSoapClient: PersonSoapClient,
+                                private val oversetter: TpsOversetter,
+                                private val pdlRestClient: PdlRestClient) {
+
+    fun hentHistorikkFor(personIdent: String, fom: LocalDate, tom: LocalDate): PersonhistorikkInfo {
+        val request = HentPersonhistorikkRequest()
+        request.aktoer = TpsPersonIdent().withIdent(NorskIdent().withIdent(personIdent))
+        request.periode = Periode().withFom(DateUtil.convertToXMLGregorianCalendar(fom))
+                .withTom(DateUtil.convertToXMLGregorianCalendar(tom))
+        val response = personSoapClient.hentPersonhistorikkResponse(request)
+        return oversetter.tilPersonhistorikkInfo(PersonIdent(personIdent), response)
+    }
+
+    fun hentPersoninfoFor(personIdent: String?): Personinfo {
+        val request: HentPersonRequest = HentPersonRequest()
+                .withAktoer(TpsPersonIdent().withIdent(NorskIdent().withIdent(personIdent)))
+                .withInformasjonsbehov(listOf(Informasjonsbehov.FAMILIERELASJONER, Informasjonsbehov.ADRESSE))
+        val response = personSoapClient.hentPersonResponse(request)
+        return oversetter.tilPersoninfo(PersonIdent(personIdent), response)
+    }
+
+    @Cacheable(cacheNames = [PERSON], key = "#personIdent", condition = "#personIdent != null")
+    fun hentPersoninfo(personIdent: String): Personinfo {
+        val request: HentPersonRequest =
+                HentPersonRequest().withAktoer(TpsPersonIdent().withIdent(NorskIdent().withIdent(personIdent)))
+                        .withInformasjonsbehov(listOf(Informasjonsbehov.FAMILIERELASJONER, Informasjonsbehov.ADRESSE))
+        val response: HentPersonResponse
+        response = personSoapClient.hentPersonResponse(request)
+        return oversetter.tilPersoninfo(PersonIdent(personIdent), response)
+    }
+
+    fun hentPersoninfo(personIdent: String,
+                       tema: String,
+                       personInfoQuery: PersonInfoQuery): Person {
+        return pdlRestClient.hentPerson(personIdent, tema, personInfoQuery)
+    }
+
+    fun hentAktørId(personIdent: String, tema: String): List<String> {
```

Comment:
Ikke-teknisk spørsmål, men; hvordan vet vi hvilken av aktørid-ene i lista som skal brukes? Er det meta-info i data-objektet her som tilsier hva som burde brukes?

Reply:
Ser at "historisk" er et attributt:
Enten bør vi returnere uten "historisk" eller så bør det følge med som metainformasjon slik at applikasjonene som skal bruke aktørid kan velge selv hvordan de skal bruke den?

Hvis du bare vil mappe fnr->aktørId så er ikke historiske så interessante, men hvis du trenger å slå opp alle id-er for en person kan det kanskje være av interesse?

Reply:
Dette endepunktet henter uten historiske forekomster. Jeg tenker at man kan lage et nytt endepunkt hvis man senere finner et behov for å hente ut historiske identer.

Reply:
Pdl leverer alt som lister. Så lenge vi kun henter aktive forekomster, så vil nok ikke listen inneholde mer enn én forekomst av hver type.